### ISSUE TITLE
Can WebView2 run without explorer.exe shell?

### ISSUE THREAD

**Issue by EdgarNZ** [MicrosoftEdge/WebView2Feedback#990]
I created a simple WinForms app with WebView2 control and it runs well in the default shell (explorer.exe) of Windows 10.

However, when I use a custom shell (not explorer.exe), it seems to fail initializing the WebView2 core component.

Is it possible to use WebView2 WinForms apps without Windows default shell?

**Comment by Syul968** [comment]
Hi @EdgarNZ,
Could you share the details about the initialization failure you are getting? Do you get an exception? Is there an error message?
Also, could you provide more information about the versions you are using?

**Version**
**SDK:** for example, 0.9.579-prerelease
**Runtime:** Browser channel and version, e.g. Canary 86.0.612.0
**Framework:** WPF, WinForms, Win32, etc.
**OS:** Win10, Win7, version info if available

**Comment by EdgarNZ** [comment]
Thank you so much for contacting me. Here is the detailed information.

A WinForms app (*CustomShell.exe*) is being used as a shell program which
is applied by changing the below registry key value.
Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows
NT\CurrentVersion\Winlogon\Shell >>>> *CustomShell.exe*
[image: image.png]

I have created a simple WinForms app (*WebView.exe*) to check compatibility
with a custom shell program.
A WebView2 control is initialized with the below codes.
[image: image.png]

* WebView.exe* works well with the shell program (explorer.exe). However,
it shows the below exception when I launch it with *CustomShell.exe*.
I have used *Process.Start()* to create it in the *CustomShell.exe* code.

[image: image.png]

*SDK:*
[image: image.png]

*Runtime: *
[image: image.png]

*Framework: WinForms - C#*
[image: image.png]

*OS:*
[image: image.png]

I hope the above information helps you investigate my issue better. Thank
you.


Regards
Edgar Kim

**Comment by Syul968** [comment]
The attached images are not showing.
Is the app trying to start the webview running as admin? If the app is running as admin the msedgewebview2.exe processes cannot run as admin and instead will run at medium IL as a child of explorer.exe.

**Comment by EdgarNZ** [comment]
Please find the attached file for the images.

I am not sure that I properly understand what you mentioned but, could you
tell me how to run msedgewebview2.exe independently (as not a child of
explorer.exe)?

> If the app is running as admin the msedgewebview2.exe processes cannot run
> as admin and instead will run at medium IL as a child of explorer.exe.


Regards
Edgar Kim

**Comment by Syul968** [comment]
Sorry for the delayed response.

The images are still not showing. Maybe GitHub email replies don't post them? Could you try copying the error text/stack in the [GitHub thread](https://github.com/MicrosoftEdge/WebView2Feedback/issues/990)? Or otherwise describe your repro steps in detail or provide a sample app?

> [...] could you tell me how to run msedgewebview2.exe independently (as not a child of explorer.exe)?

To run the WebView2 Runtime processes as children of your app (and not explorer.exe) your app must run without admin privileges. Are you running your application with admin privileges when hitting this issue?

We are seeing other reports like this (#960 #343) and we are planning to investigate this soon. A stack/error code would be really helpful.

**Comment by EdgarNZ** [comment]
Thank you for your explanation.
I realised that my issue occured because I had tried to launch the webview2
app via my custom shell that has admin privileges.

I could see two options to avoid this issue.

One is to run my custom shell app as a standard user and launch the
webview2 app from it.
Another is to launch the webview2 app by using standard user credentials.
It works even though my custom shell has admin privileges.

Thank you so much for your help.

Regards
Edgar Kim
